language/compiler/src/lib.rs

```diff
@@ -77,35 +66,23 @@ impl Compiler {
     }
 
     fn compile_script(
-        &mut self,
+        self,
         file_name: &str,
         code: &str,
     ) -> Result<(CompiledScript, SourceMap<Loc>, Vec<CompiledModule>)> {
         let parsed_script = parse_script(file_name, code)?;
-        let deps = self.deps();
-        let (compiled_script, source_map) = compile_script(None, parsed_script, &deps)?;
-        Ok((compiled_script, source_map, deps))
+        let (compiled_script, source_map) = compile_script(None, parsed_script, &self.deps)?;
```

Comment:
Excellent! I love how you were able to remove `mut` in all these functions.

Reply:
Yeah I think it's even possible for us to go one step further and delete this interface, or even the `compiler` crate as a whole. IMO clients can just use the interfaces from `ir-to-bytecode` directly. But let's do that in a separate PR.

Reply:
@tnowacki opinions on getting rid of the compiler crate (and/or the IR compiler binary)?

Reply:
I would prefer to keep the IR compiler binary as long as we've got tests written in the IR language. The plan we had discussed in the past was to incrementally mutate that compiler into a bytecode assembler, and that still sounds good to me.

Reply:
Yeah I wouldn't really touch this much until things change. No point cleaning it up or doing anything crazy until we add the bytecode syntax. Then we will likely want a tool that round-trips with the disassembler